test/controllers/user_controller_test.exs

```diff
@@ -91,20 +95,14 @@ defmodule CodeCorps.UserControllerTest do
       assert user.id == slugged_route.user_id
     end
 
-    test "calls segment tracking after user is created", %{conn: conn} do
-      conn = post conn, user_path(conn, :create), %{
-        "meta" => %{},
-        "data" => %{
-          "type" => "user",
-          "attributes" => Map.put(@valid_attrs, :password, "password"),
-          "relationships" => relationships
-        }
-      }
-      id = json_response(conn, 201)["data"]["id"] |> String.to_integer
-      assert_received {:track, ^id, "Signed Up", %{}}
+    test "calls segment tracking", %{conn: conn} do
+      id = json_response(conn, 201)["data"]["id"]
+      assert_received {:track, id, "Signed Up", %{}}
```

Comment:
My guess is, the compiler is telling you the `variable id is defined but not used`.  The reason for that is that this call actually does not check if the second tupple element is equal to `id`. Instead, it redefines the `id` variable altogether.

The correct call would be 

```Elixir
assert_received {:track, ^id, "Signed Up", %{}}
```

My guess is, you took this from another part of our code. Sadly, it was wrong in our code and I since corrected it. Sorry about that.